### PR TITLE
Rotate images pull request

### PR DESCRIPTION
**Pull request by tsdimobile**
- we created a new file preview action menu entry called "Rotate right" (below menu entry "Details") with rotate icon (see attached zip)
- we implemented rotate function:
- after taping on rotate menu element, the image rotates 90 degrees clockwise
- Image is resized to fill the viewport in either width or height (whichever is reached first)
- we saved the rotated image to the server

Simulator Screen Shot - iPhone 12 - 2021-11-01 at 13 05 23

### CODE REVIEW

**Comment by tobiasKaminsky** [comment]
Is this shown also in file list?
Or only when user actually is in photo preview and then clicks "..."? <- I prefer this, as it does not clutter the option menu.

**Comment by tsdimobile** [comment]
@tobiasKaminsky rotate image option is added only when user clicks on 3 .dot menu when previews the image, we haven't added this in option Menu.

**Review by thisIsTheFoxe — CHANGES_REQUESTED** [review]
Hi, and thank you for your contribution, what a nice idea.
Looks good so far.

Is there a reason why this can't be added to `NCViewerQuickLook` instead? Like a simple navigation item next to the markup one? Another advantage would be that you could rotate multiple times, without saving and uploading every single time, just when you're *done*.
Also, I noticed that after rotating the image doesn't update, it just vanishes. Would be great to have a better UX. If nothing else, at least make sure the detail view pops back.

Thanks again :)

File: iOSClient/ScanDocument/ScanCollectionView.swift [hunk 1]
```diff
@@ -477,6 +477,9 @@ extension DragDropViewController : UICollectionViewDataSource {
 
 extension UIImage {
     func rotate(radians: Float) -> UIImage? {
+        if self.size.height == 0.0 || self.size.width == 0.0 {
```
[the remark this answers is not in the document]
Comment (harshaltsystems) [reply iOSClient/ScanDocument/ScanCollectionView.swift:478]: Fixed

File: iOSClient/Viewer/NCViewerMedia/NCViewerMediaZoom.swift [hunk 2]
```diff
@@ -286,6 +287,39 @@ class NCViewerMediaZoom: UIViewController {
...
     }
+    
+    @objc func rotateImage() {
+
+//        imageVideoContainer.image = image
+//        imageVideoContainer.sourceImage = image
+//        let rotatedImage = imageView.image?.rotate(radians: .pi/2)
+
+        let rotatedImage = imageVideoContainer.image?.rotate(radians: .pi/2)
+        if rotatedImage == nil {
```
[the remark this answers is not in the document]
Comment (harshaltsystems) [reply iOSClient/Viewer/NCViewerMedia/NCViewerMediaZoom.swift]: Fixed

**Comment by harshaltsystems** [comment]
Rotate image functionality changed 
- Confirmation dialogue added before leaving the rotated imaged screen.
- If user confirms the changes then request for rotate image proceeds.

**Comment by thisIsTheFoxe** [comment]
Hi @harshaltsystems,
Thanks a lot for all your work and for keeping this PR active!
Please note that Nextcloud enforced the [Developer Certificate of Origin (DCO)](https://github.com/thisIsTheFoxe/ios#creating-pull-requests) on pull requests. Make sure that you know what that means and confirm it by signing off all your commits. E.g. like this:
```
git rebase HEAD~6 --signoff
git push --force-with-lease origin pr_rotate_image
```
<br/>

> Is there a reason why this can't be added to NCViewerQuickLook instead? Like a simple navigation item next to the markup one? Another advantage would be that you could rotate multiple times, without saving and uploading every single time, just when you're done.

Also, feel free to let me know what you think about this approach..? I think this would have a much more intuitive UX and avoids cluttering up the menus..

**Pull request closed without merging**